internal/workflow/ProjectAPIKeyOperator.java

```diff
@@ -30,41 +36,42 @@ public UUID createProjectAPIKey(UUID projectId, String description) throws Excep
         return newApiKeyId;
     }
 
-    public UUID getOldKeyUUID(UUID projectId, String oldKey) throws Exception {
+    public SimpleEntry<AuthToken, UUID> getProjectAuthTokenByToken(String token) throws Exception {
+        ProjectList projectList = projectsApi.findProjects(null, null, null, null);
+        if (projectList.getProjects().size() != 1) {
+            throw new Exception("Expected only one project to be associated with the token");
+        }
+        UUID projectId = projectList.getProjects().get(0).getId();
+
         AuthTokenList tokenList = authApi.findProjectAPIKeys(projectId, null, null);
-        for (AuthToken token : tokenList.getApiKeys()) {
-            if (token.getToken().equals(oldKey)) {
-                return token.getId();
+        for (AuthToken authToken : tokenList.getApiKeys()) {
+            if (authToken.getToken().equals(token)) {
+                return new SimpleEntry<>(authToken, projectId);
             }
         }
-        throw new Exception("Old API key not found");
+        throw new Exception("Provided token not found");
     }
 
     public void deleteAPIKey(UUID apiKeyId) throws Exception {
         authApi.deleteAPIKey(apiKeyId);
     }
 
-    public static void main(String[] args) {
-        String oldApiKey = "";
-        if (oldApiKey.isEmpty()) {
-            throw new IllegalArgumentException("API Key is not set. Please set the oldApiKey variable.");
-        }
+    public String rotateProjectKey(String authToken) {
+        try {
+            SimpleEntry<AuthToken, UUID> result = getProjectAuthTokenByToken(authToken);
+            AuthToken oldAuthToken = result.getKey();
+            UUID projectId = result.getValue();
 
-        String projectIdString = "";
-        if (projectIdString.isEmpty()) {
-            throw new IllegalArgumentException("Project ID is not set. Please set the projectID variable.");
-        }
+            if (oldAuthToken == null || oldAuthToken.getProject() == null) {
+                throw new Exception("Cannot rotate API key. The current API key is not associated with a project.");
+            }
 
-        ProjectAPIKeyOperator projAPIKeyOperator = new ProjectAPIKeyOperator(oldApiKey);
-        UUID projectId = UUID.fromString(projectIdString);
-        try {
-            projAPIKeyOperator.createProjectAPIKey(projectId, "test-api-key");
+            UUID newApiKeyId = createProjectAPIKey(projectId, "rotated-api-key");
+            deleteAPIKey(oldAuthToken.getId());
 
-            UUID oldApiKeyId = projAPIKeyOperator.getOldKeyUUID(projectId, oldApiKey);
-            projAPIKeyOperator.deleteAPIKey(oldApiKeyId);
+            return newApiKeyId.toString();
```

Comment:
This may be subjective, but I think the return value here could be improved.

If this is going to return a string, it should be the new token.
Otherwise, it seems this would return a UUID (the UUID of the new token).

Another option would be to return the `newAuthToken` here and in `createProjectAPIKey`. The tests would need to be updated to match.

Reply:
modified to return the auth token instead of the ID. Updated the Test class accordingly.